x-pack/plugins/ml/public/timeseriesexplorer/timeseriesexplorer_controller.js

```diff
@@ -136,91 +148,90 @@ module.controller('MlTimeSeriesExplorerController', function (
+      const invalidIds = _.difference(selectedJobIds, timeSeriesJobIds);
+      selectedJobIds = _.without(selectedJobIds, ...invalidIds);
+      if (invalidIds.length > 0) {
+        let warningText = i18n('xpack.ml.timeSeriesExplorer.canNotViewRequestedJobsWarningMessage', {
+          defaultMessage: `You can't view requested {invalidIdsCount, plural, one {job} other {jobs}} {invalidIds} in this dashboard`,
+          values: {
+            invalidIdsCount: invalidIds.length,
+            invalidIds
           }
-
-          if (selectedJobIds.length > 1 || mlJobSelectService.groupIds.length) {
-          // if more than one job or a group has been loaded from the URL
-            if (selectedJobIds.length > 1) {
-            // if more than one job, select the first job from the selection.
-              toastNotifications.addWarning(
-                i18n('xpack.ml.timeSeriesExplorer.youCanViewOneJobAtTimeWarningMessage', {
-                  defaultMessage: 'You can only view one job at a time in this dashboard'
-                })
-              );
-              mlJobSelectService.setJobIds([selectedJobIds[0]]);
-            } else {
-            // if a group has been loaded
-              if (selectedJobIds.length > 0) {
-              // if the group contains valid jobs, select the first
-                toastNotifications.addWarning(
-                  i18n('xpack.ml.timeSeriesExplorer.youCanViewOneJobAtTimeWarningMessage', {
-                    defaultMessage: 'You can only view one job at a time in this dashboard'
-                  })
-                );
-                mlJobSelectService.setJobIds([selectedJobIds[0]]);
-              } else if ($scope.jobs.length > 0) {
-              // if there are no valid jobs in the group but there are valid jobs
-              // in the list of all jobs, select the first
-                mlJobSelectService.setJobIds([$scope.jobs[0].id]);
-              } else {
-              // if there are no valid jobs left.
-                $scope.loading = false;
-              }
-            }
-          } else if (invalidIds.length > 0 && selectedJobIds.length > 0) {
-          // if some ids have been filtered out because they were invalid.
-          // refresh the URL with the first valid id
-            mlJobSelectService.setJobIds([selectedJobIds[0]]);
-          } else if (selectedJobIds.length > 0) {
-          // normal behavior. a job ID has been loaded from the URL
-            loadForJobId(selectedJobIds[0]);
+        });
+        if (selectedJobIds.length === 0 && timeSeriesJobIds.length > 0) {
+          warningText += i18n('xpack.ml.timeSeriesExplorer.autoSelectingFirstJobText', {
+            defaultMessage: ', auto selecting first job'
+          });
+        }
+        toastNotifications.addWarning(warningText);
+      }
+      // TODO: removed -> || mlJobSelectService.groupIds.length -> replace functionality
```

Comment:
If you are leaving this section in, the indentation on a lot of the comments looks a bit out.